### DISCUSSION TITLE
We've added price descriptors to Intervals, and network name and tariff codes to sites

### DISCUSSION THREAD

**Discussion by madpilot** [amberelectric/public-api#100] [Announcements]
# Price descriptors

When I originally designed the API, I made an explicit decision to omit the price descriptions we use to drive the colours in that app, assuming that people would just set their own.

Turns out, that assumption was wrong, and there has been a number of requests for it, so now you can see the price descriptor that we use internally to drive out colours!

```json
{
    "type": "ForecastInterval",
    "date": "2022-02-22",
    "duration": 30,
    "startTime": "2022-02-22T06:30:01Z",
    "endTime": "2022-02-22T07:00:00Z",
    "nemTime": "2022-02-22T17:00:00+10:00",
    "perKwh": 17.0241,
    "renewables": 38.20271932892017,
    "spotPerKwh": 2.63034,
    "channelType": "general",
    "spikeStatus": "none",
    "descriptor": "low",
    "range": {
      "min": 0.895,
      "max": 13.89635
    }
}
```

The values are `negative`, `veryLow`, `low`, `neutral`, `high` and `spike`

In the app, `negative` is green with an exclamation mark, `veryLow` is green, `low` is yellow, `neutral` is orange, `high` is red and `spike` is maroon.

# Additional site details

I've also added a couple of extra attributes on the `Site` and `Channel` objects.

`Site` now includes the network name, and each channel has its tariff code. While we aren't likely to dive deeper into breaking apart tariffs, this will make it a bit easier to find the relevant documents so you can find the underlying tariff information yourself.

**Comment by GrahamLea** [comment]
Great stuff!

Just for clarity, I assume the `descriptor` is not related to custom notifications, and is purely Amber's rating of the price?

**Reply by madpilot** [reply]
Correct. It's the thing that drives the colours in the app.

**Comment by scotta1725** [comment]
Thanks for that Myles, a great addition.  Really impressed at how proactive you and Amber Electric is. 
Will you also be looking to update the Home Assistant Integration at some stage to support it?

**Reply by madpilot** [reply]
Yep - Hopefully by the end of the week.

**Comment by madpilot** [comment]
I'm going to make a small (technically breaking) change, and adding an `extremelyLow` value, as we are tweaking the levels internally, and `negative` is now misleading.

If you have started using that field, treat `extremelyLow` the same as `negative`

**Reply by madpilot** [reply]
I can hold off if people already have loadbearing code relying on this, though I suspect the surface area for this is small at the moment, given we only released that change last week, and most of you are waiting for me to update the home assistant integrations 😆

**Reply by madpilot** [reply]
This is live now

**Comment by madpilot** [comment]
`negative` is still valid, it just won't ever be used, and will be removed in the future (Docs will be updated to reflect that)
